### ISSUE TITLE
[BUG] OAS -> Protobuf generates erroneous fieldnames with "Underscore" in the name instead of real _

### ISSUE THREAD

**Issue by aantono** [OpenAPITools/openapi-generator#9544]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

Given the following OASv3 Schema:

```yaml
components:
  schemas:
    Test:
      type: object
      properties:
        local_date_time:
          $ref: '#/components/schemas/LocalDateTime'
        offset_date_time:
          $ref: '#/components/schemas/OffsetDateTime'
        zoned_date_time:
          $ref: '#/components/schemas/ZonedDateTime'
        big_decimal:
          $ref: '#/components/schemas/BigDecimal'
    LocalDateTime:
      type: string
      format: 'date-time'
      pattern: '\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+'
    OffsetDateTime:
      type: string
      format: 'date-time'
      pattern: '\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)'
    ZonedDateTime:
      type: string
      format: 'date-time'
      pattern: '\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+\[[a-z/]+\]'
    BigDecimal:
      type: string
      pattern: '\d+(\.\d+)?'
info:
  title: ''
  version: v1beta1
openapi: 3.0.3
paths: {}
```

when executing the following command:

`openapi-generator generate -i test.yaml -g protobuf-schema -o out`

the generated output looks like this:

```proto
/*
  No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 
  The version of the OpenAPI document: v1beta1
  
  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package ;


message Test {

  string localUnderscoredateUnderscoretime = 399350043;

  string offsetUnderscoredateUnderscoretime = 141471169;

  string zonedUnderscoredateUnderscoretime = 137492891;

  string bigUnderscoredecimal = 226570700;

}
```

As evident from the field names, they all contain `Underscore` verbatim spelled out, instead of having an actual `_` character.

The expected correct result should be:

```proto
/*
  No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 
  The version of the OpenAPI document: v1beta1
  
  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package ;


message Test {

  string local_date_time = 399350043;

  string offset_date_time = 141471169;

  string zoned_date_time = 137492891;

  string big_decimal = 226570700;

}
```

```
openapi-generator version
5.1.1
```

**Comment by blackplane** [comment]
This name conversion is done [here](https://github.com/OpenAPITools/openapi-generator/blob/43963602ce3cd946ed1080240b01fe4cf674536e/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GrpcSchemaCodegen.java#L203) by calling `camelize`. This method is defined [here](https://github.com/OpenAPITools/openapi-generator/blob/158649aff9a87b13b448880d7ca6a4ac35fadcc2/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/StringUtils.java#L168).

So, per default, it converts everything (including the underscore) to CamelCase. The correct behavior would be to accept the

**Comment by imzeali** [comment]
Add this option "--additional-properties=identifierNamingConvention=snake_case" , still doesn't work

**Comment by clsx524** [comment]
I don't think identifierNamingConvention can be applied to protobuf-schema at this moment. I have to do a global replacement of `Underscore` to `_` but obviously it's no ideal.

**Comment by wing328** [comment]
Closed via #13296 

Please pull the latest master to give it a try.
